Packs/OpenAI/Integrations/OpenAI/OpenAI.py

```diff
@@ -1,109 +1,230 @@
+            "presence_penalty": presence_penalty,
+            "best_of": best_of,
+            "stop": stop
+        }
+        if self.is_azure:  # Azure OpenAI
+            url_suffix = f"openai/deployments/{model}/completions?api-version={self.version}"
+        else:  # standard api.openai.com
+            data["model"] = model
+            url_suffix = "v1/completions"
+        return self._http_request(
+            method='POST', url_suffix=url_suffix, json_data=data,
+            headers=self.headers, resp_type='json', ok_codes=(200,)
+        )
+
+    def chatgpt(self, prompt: str, model: str = CHAT_DEFAULT_MODEL) -> dict:
+        """
+            Send prompt to ChatGPT using the 'chat/completions' endpoint.
+
+            Args:
+                prompt (str): Input to ChatGPT
+                model (str): The model which will generate the chat response
+
+            Returns:
+                dict: HTTP response from the OpenAI Chat API
+        """
+        options = {
+            "max_tokens": 1000,
+            "messages": [{
+                "role": "user",
+                "content": prompt
+            }]
         }
+        if self.is_azure:  # Azure OpenAI
+            url_suffix = f"openai/deployments/{model}/chat/completions?api-version={self.version}"
+        else:  # standard api.openai.com
+            options["model"] = model
+            url_suffix = "v1/chat/completions"
+        return self._http_request(
+            method='POST', url_suffix=url_suffix,
+            json_data=options, headers=self.headers,
+            resp_type="json", ok_codes=(200,)
+        )
+
+
+''' HELPER FUNCTIONS '''
+
 
-        return self._http_request(method='POST', url_suffix='v1/completions', json_data=data, headers=self.headers,
-                                  resp_type='json', ok_codes=(200,), )
+def chatgpt_output(response) -> CommandResults:
+    """
+        Convert response from ChatGPT to a human readable format in markdown table
+
+        :return: CommandResults return output of ChatGPT response
+        :rtype: ``CommandResults``
+    """
+    if response and isinstance(response, dict):
+        rep = json.dumps(response)
+        repJSON = json.loads(rep)
+        model = repJSON.get('model')
+        createdTime = repJSON.get('created')
+        id = repJSON.get('id')
+        choices = repJSON.get('choices', [])[0].get('message', {}).get('content', "").strip('\n')
+        promptTokens = repJSON.get('usage', {}).get('prompt_tokens')
+        completionTokens = repJSON.get('usage', {}).get('completion_tokens')
+        totalTokens = repJSON.get('usage', {}).get('total_tokens')
+        context = [{'id': id, 'Model': model,
+                    'ChatGPT Response': choices, 'Created Time': createdTime,
+                    'Number of Prompt Tokens': promptTokens,
+                    'Number of Completion Tokens': completionTokens,
+                    'Number of Total Tokens': totalTokens
+                    }]
```

Comment:
```suggestion
        context = [{'id': id, 'Model': model,
                    'ChatGPT_Response': choices, 'Created Time': createdTime,
                    'Number_of_Prompt_Tokens': promptTokens,
                    'Number_of_Completion_Tokens': completionTokens,
                    'Number_of_Total _Tokens': totalTokens
                    }]
```
It is not recommended to use spaces in the context path.

Reply:
I'll check with my team and let you know

Reply:
After checking, it may cause bugs if the context outputs will remain with spaces, so it is better to change them.